recipes/server.rb

```diff
@@ -197,33 +212,74 @@ def remove_job(job, servernode)
+        deep_merge!(job, n['rdiff-backup']['client']['fs']['jobs'][src] || {}) # Merge the client's job-specific attributes over the top. (Levels 8 and 9)
+
+        job['type'] = 'fs'
+        # Keep higher-level attributes in the job object for convenience.
+        job['source'] = src
+        job['fqdn'] = n['fqdn']
+        job['name'] = "#{job['type']}_#{job['fqdn']}_#{job['source'].gsub("/", "-")}" # Name is used for referencing this job and is unique.
+        job['user'] = n['rdiff-backup']['client']['user']
+        job['ssh-port'] = n['rdiff-backup']['client']['ssh-port']
+        job['ssh-key'] = n['rdiff-backup']['client']['fs']['ssh-key']
+
+        # Remove exclusion rules that don't apply to this job
+        relevantdirs = []
+        job['exclude-dirs'].each do |dir|
+          relevantdirs << dir if dir.start_with?(src) or dir.start_with?('*')
+        end
+        job['exclude-dirs'] = relevantdirs
+
+        jobs << job
+      end
+    end
+  end
 
-      job = deep_copy(servernode['rdiff-backup']['server']['jobs']['default']) # Start with the server's default attributes. (Levels 1, 2, and 3)
-      deep_merge!(job, n['rdiff-backup']['client']['jobs']['default'] || {}) # Merge the client's default attributes over the top. (Levels 4 and 5)
-      deep_merge!(job, servernode['rdiff-backup']['server']['jobs'][src] || {}) # Merge the server's job-specific attributes over the top. (Levels 6 and 7)
-      deep_merge!(job, n['rdiff-backup']['client']['jobs'][src] || {}) # Merge the client's job-specific attributes over the top. (Levels 8 and 9)
+  mysqldbs = Set.new
+  if servernode['rdiff-backup']['server']['mysql']['enable'] and n['rdiff-backup']['client']['mysql']['enable']
+    mysqldbs.merge(servernode['rdiff-backup']['server']['mysql']['jobs'].keys)
+    mysqldbs.merge(n['rdiff-backup']['client']['mysql']['jobs'].keys)
+
+    # If there are no databases specified, create jobs for each database.
+    if mysqldbs.empty?
+      include_recipe 'mysql::client'
+      #TODO: Use mysql2 gem instead of shelling out?
+      #chef_gem 'mysql2'
+      #mysqlconn = Mysql2::Client.new(:host => n['fqdn'], :username => "rdiff-backup")
+      #mysqldbs = mysqlconn.query("SHOW DATABASES")
+      command = "ssh -i #{File.join('/home', servernode['rdiff-backup']['server']['user'], '.ssh', 'id_rsa')} -p #{n['rdiff-backup']['client']['ssh-port']} #{n['rdiff-backup']['client']['user']}@#{n['fqdn']} mysql -e 'SHOW\\ DATABASES' -u #{n['rdiff-backup']['client']['mysql']['mysql-user']} -p'#{n['rdiff-backup']['client']['mysql']['mysql-password']}' --column-names=0"
+      mysqldbs = `#{command}`.split()
+      if mysqldbs.empty?
+        Chef::Log.warn("Unable to connect to database '#{n['rdiff-backup']['client']['mysql']['mysql-user']}@#{n['fqdn']}'")
+      end
+    end
 
+    mysqldbs.each do |db|
+      puts "DEBUG: db"
```

Comment:
I was just putting that there for short-term debugging.  I'm going to remove it now.